src/jit/flowgraph.cpp

```diff
@@ -22995,6 +22995,7 @@ void Compiler::fgRemoveEmptyTry()
                     fgRemoveStmt(block, finallyRet);
                     block->bbJumpKind = BBJ_ALWAYS;
                     block->bbJumpDest = continuation;
+                    fgAddRefPred(continuation, block);
                 }
```

Comment:
Looks like I need to rebase to get past a merge conflict, so I'll also add the assert.